doc/elements-tx-format.md

```diff
@@ -0,0 +1,570 @@
+
+Serialized format:
+`[Issuance Amount Range Proof][Inflation Keys Range Proof][Script Witness][Pegin Witness]`
+
+The range proofs must be empty if their asociated amounts (issuance / inflation keys) are explicit.
+
+#### OutputWitness
+
+|Field|Required|Size|Data Type|Encoding|Notes| 
+|-----|--------|-----|---------|--------|-----|
+| Surjection Proof | Yes | Varies | `Proof` | | Will be null (`0x00`) if corresponding output is not confidential. |
+| Range Proof | Yes | Varies | `Proof` | | Will be null (`0x00`) if corresponding output is not confidential. |
+
+Serialized format: `[Surjection Proof][Range Proof]`
+
+The Range Proof must be empty if the output’s amount is explicit. 
+
+The Surjection Proof must be empty if the output’s assetId is explicit.
+
+#### AssetIssuance
+
+|Field|Required|Size|Data Type|Encoding|Notes| 
+|-----|--------|-----|---------|--------|-----|
+| Asset Blinding Nonce | Yes | 32 bytes | `hex` | | Zero for a new asset issuance; otherwise a blinding factor for the input. |
+| Asset Entropy | Yes | 32 bytes | `hex` | | **New issuances:** Freeform entropy field, no consensus-defined meaning, but is used as additional entropy to the asset tag calculation.<br><br>**Reissuances:** Required to be the asset's entropy value (from its initial issuance). |
+| Amount | Yes | 1 or 9 or 33 bytes | `ConfidentialAmount` | | Amount of the asset to issue. Both explicit and blinded amounts are supported. |
+| Num Inflation Keys | Yes | 1 or 9 or 33 bytes | `ConfidentialAmount` | | Number of inflation keys to issue. Both explicit and blinded amounts are supported.<br><br>**Note:** inflation keys cannot be reissued. |
+
+Serialized format: `[Asset Blinding Nonce][Asset entropy][Amount][Num Inflation Keys]`
+
+#### Variable Length Integer (VarInt)
+
+This data type is derived from Bitcoin and allows an integer to be encoded depending on the represented value, to save space. Variable length integers always precede an vector of a type of data that may vary in length. Longer numbers are encoded in little-endian.
+
+Refer to the [Bitcoin protocol documentation](https://en.bitcoin.it/wiki/Protocol_documentation#Variable_length_integer) for a full description on how this type is serialized.
+
+#### Vector\<Type\>
+
+Serialized format: `[Length (n)][Item #1][Item #2][...][Item #n]`
+
+Each `Vector` begins with a `VarInt` describing the number of items it contains.
+
+If the vector is of type `hex`, then the size / structure of each individual item is not known in advance. In this case, each item begins with a `VarInt` describing its size `s` in bytes, followed by `s` bytes which should be interpreted as the item itself.
+Otherwise, size prefixes are omitted, and each item should be interpreted in accordance with the vector's type.
+
+Refer to the examples section below for more concrete examples of serialized vectors.
+
+#### ConfidentialAsset
+
+Serialized format: `[header][value]`
```

Comment:
might be helpful to keep with the table description for these types, for consistency and clarity (even though there are only 2 fields).